parsl/executors/high_throughput/executor.py

```diff
@@ -0,0 +1,403 @@
+            }
+
+        We do not support these yet, but they could be added easily.
+
+        .. code:: python
+
+            {
+               "task_id" : <task_id>
+               "cpu_stat" : <>
+               "mem_stat" : <>
+               "io_stat"  : <>
+               "started"  : tstamp
+            }
+
+        The `None` message is a die request.
+        """
+        logger.debug("[MTHREAD] queue management worker starting")
+        while True:
+            try:
+                msg = self.incoming_q.get(timeout=1)
+                logger.debug("[MTHREAD] get has returned")
+
+            except queue.Empty as e:
+                logger.debug("[MTHREAD] queue empty")
+                # Timed out.
+                pass
+
+            except IOError as e:
+                logger.exception("[MTHREAD] Caught broken queue with exception code {}: {}".format(e.errno, e))
+                return
+
+            except Exception as e:
+                logger.exception("[MTHREAD] Caught unknown exception: {}".format(e))
+                return
+
+            else:
+
+                if msg is None:
+                    logger.debug("[MTHREAD] Got None, exiting")
+                    return
+
+                else:
+                    try:
+                        tid = msg['task_id']
+                    except Exception as e:
+                        raise BadMessage("Message received does not contain 'task_id' field")
+
+                    task_fut = self.tasks[tid]
+
+                    if 'result' in msg:
+                        result, _ = deserialize_object(msg['result'])
+                        task_fut.set_result(result)
+
+                    elif 'exception' in msg:
+                        try:
+                            exception, _ = deserialize_object(msg['exception'])
+                            task_fut.set_exception(exception)
+                        except Exception as e:
+                            # TODO could be a proper wrapped exception?
+                            task_fut.set_exception(
+                                DeserializationError("Received exception, but handling also threw an exception: {}".format(e)))
+                    else:
+                        raise BadMessage("Message received is neither result or exception")
+
+            if not self.is_alive:
+                break
+        logger.info("[MTHREAD] queue management worker finished")
+
+    # When the executor gets lost, the weakref callback will wake up
+    # the queue management thread.
+    def weakref_cb(self, q=None):
+        """We do not use this yet."""
+        q.put(None)
+
+    def _start_local_queue_process(self):
+
+        comm_q = Queue(maxsize=10)
+        self.queue_proc = Process(target=interchange.starter,
+                                  args=(comm_q,),
+                                  kwargs={"client_ports": (self.outgoing_q.port, self.incoming_q.port),
```

Comment:
We are not passing the `logging_level` kwarg. I think it would be fine to add
```
"logging_level": logging.DEBUG if self.worker_debug else logging.INFO
```
here (i.e. have the interchange logging level be debug if `worker_debug` is `True`), but if you want more granularity we could add another knob for the interchange debug level.

Reply:
Fixed.